code/renderers/svelte/src/components/PreviewRender.svelte

```diff
@@ -30,19 +36,17 @@
         Component,
         props,
         on,
+        isOriginalStory,
       };
     }
     return storyFn();
   }
 
   // reactive, re-render on storyFn change
-  $: ({ Component, props = {}, on } = getStoryFnValue(storyFn));
+  $: ({ Component, props = {}, on, isOriginalStory } = getStoryFnValue(storyFn));
 
-  const eventsFromArgTypes = Object.fromEntries(
-    Object.entries(storyContext.argTypes)
-      .filter(([k, v]) => v.action && props[k] != null)
-      .map(([k, v]) => [v.action, props[k]])
-  );
+  // set the argTypes context, read by the last SlotDecorator that renders the original story
```

Comment:
Why do you need this to be context? The SlotDecorator could get the argTypes just in decorators.ts right? As an extra prop to SlotDecorator

Reply:
https://github.com/storybookjs/storybook/blob/1b17dfee562a85be19013071f1d7703b945d9b2a/code/renderers/svelte/src/decorators.ts#L62-L69

Reply:
good call. fixed!